### PR TITLE
Feat/us041 authorisation system

### PR DESCRIPTION
**Pull request by DariuszMech**

### CODE REVIEW

File: app.log [hunk 1]
```diff
@@ -0,0 +1,101 @@
+2024-08-28 10:56:41.447 [main] INFO  org.hibernate.validator.internal.util.Version - HV000001: Hibernate Validator 6.1.2.Final
```
Comment (Alicja-Kainos) [inline app.log]: remove logs from PR

File: src/main/java/org/example/auth/JwtAuthenticator.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import io.dropwizard.auth.Authenticator;
+import io.jsonwebtoken.Jwts;
+import org.example.models.JwtToken;
+import org.example.models.UserRole;
+
+import java.security.Key;
+import java.util.Optional;
+
+public class JwtAuthenticator implements Authenticator<String, JwtToken> {
+    Key key;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/auth/JwtAuthenticator.java]: private  and getter?

File: src/main/java/org/example/controllers/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                                name = HttpHeaders.AUTHORIZATION,
+                                in = ApiKeyAuthDefinition.ApiKeyLocation.HEADER
+                        )
+                }
+        )
+)
+
+public class AuthController {
+
+    AuthService authService;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/controllers/AuthController.java]: private

File: src/main/java/org/example/models/JwtToken.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package org.example.models;
+
+import javax.security.auth.Subject;
+import java.security.Principal;
+
+public class JwtToken implements Principal {
+
+    UserRole userRole;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/JwtToken.java]: private

File: src/main/java/org/example/models/LoginRequest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package org.example.models;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LoginRequest {
+
+    String email;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/LoginRequest.java]: private

File: src/main/java/org/example/models/LoginRequest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    String password;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/LoginRequest.java]: private

File: src/main/java/org/example/models/User.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
+package org.example.models;
+
+public class User {
+    final String email;
+    final String password;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/User.java]: priavates

File: src/main/java/org/example/models/UserRole.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package org.example.models;
+
+import java.util.Map;
+
+public class UserRole {
+
+    public static final String ADMIN = "admin";
+    public static final String USER = "user";
+    int id;
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/models/UserRole.java]: private

**Review by Alicja-Kainos — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**